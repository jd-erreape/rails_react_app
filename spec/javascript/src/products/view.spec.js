import React from 'react';
import { shallow } from 'enzyme';
import { Table } from 'react-bootstrap';
import View from 'src/products/view';  

const products = [
  {
    code: 'CD1',
    name: 'Product1',
    price: 100,
  },
  {
    code: 'CD2',
    name: 'Product2',
    price: 200,
  },
];

describe('View component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<View products={products} />);

    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('contains proper number of products', () => {
    const wrapper = shallow(<View products={products} />);

    expect(wrapper.find('tbody tr').length).toBe(2)
  });
});