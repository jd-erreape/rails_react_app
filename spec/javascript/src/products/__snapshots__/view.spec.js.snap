// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View component matches snapshot 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        Code
      </th>
      <th>
        Name
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="CD1"
    >
      <td>
        CD1
      </td>
      <td>
        Product1
      </td>
      <td>
        100
      </td>
    </tr>
    <tr
      key="CD2"
    >
      <td>
        CD2
      </td>
      <td>
        Product2
      </td>
      <td>
        200
      </td>
    </tr>
  </tbody>
</table>
`;
